salesforce_logs_receiver.go

```diff
@@ -0,0 +1,193 @@
+	}, nil
+}
+
+func (slr *SalesforceLogsReceiver) LoginSalesforce() error {
+	if err := slr.client.LoginPassword(slr.username, slr.password, slr.securityToken); err != nil {
+		return fmt.Errorf("error login Salesforce API: %w", err)
+	}
+
+	return nil
+}
+
+func (slr *SalesforceLogsReceiver) CollectSObject(sObjectName string) error {
+	query := fmt.Sprintf("SELECT Id,CreatedDate FROM %s", sObjectName)
+	result, err := slr.client.Query(query)
+	if err != nil {
+		return fmt.Errorf("error querying Salesforce API: %w", err)
+	}
+
+	for _, record := range result.Records {
+		id := record.StringField("Id")
+		data := record.Get(id)
+
+		var jsonData []byte
+		jsonData, err = json.Marshal(data)
+		if err != nil {
+			return fmt.Errorf("error marshaling data from Salesforce API: %w", err)
+		}
+
+		if strings.ToLower(sObjectName) == eventLogFileSObjectName {
+			jsonData, err = slr.handleEventLogFileSObject(data, jsonData)
+			if err != nil {
+				return fmt.Errorf("error handling event log file sObject: %w", err)
+			}
+		}
+
+	}
+
+	return nil
+}
+
+func (slr *SalesforceLogsReceiver) handleEventLogFileSObject(data *simpleforce.SObject, jsonData []byte) ([]byte, error) {
+	eventLog, err := slr.getEventLogFileContent(data)
+	if err != nil {
+		return nil, fmt.Errorf("error getting event log file content: %w", err)
+	}
+
+	jsonData, err = addEventLogToJsonData(eventLog, jsonData)
+	if err != nil {
+		return nil, fmt.Errorf("error adding event log to JSON data: %w", err)
+	}
+
+	return jsonData, nil
+}
+
+func (slr *SalesforceLogsReceiver) getEventLogFileContent(data *simpleforce.SObject) (map[string]interface{}, error) {
+	apiPath := data.StringField("LogFile")
+	logFileContent, err := slr.getFileContent(apiPath)
+	if err != nil {
+		return nil, fmt.Errorf("error getting event log file content: %w", err)
+	}
+
+	reader := strings.NewReader(string(logFileContent))
+	csvReader := csv.NewReader(reader)
+
+	csvData, err := csvReader.ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("error reading CSV data: %w", err)
+	}
+
+	logEvent := make(map[string]interface{}, 0)
+	for index, value := range csvData {
+		if index == 0 {
+			continue
+		}
+
+		key := csvData[0][index-1]
+		logEvent[key] = value[index-1]
+	}
+
+	return logEvent, nil
+}
+
+func (slr *SalesforceLogsReceiver) getFileContent(apiPath string) ([]byte, error) {
+	httpClient := &http.Client{}
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s%s", strings.TrimRight(slr.client.GetLoc(), "/"), apiPath), nil)
+	req.Header.Add("Content-Type", "application/json; charset=UTF-8")
+	req.Header.Add("Accept", "application/json")
+	req.Header.Add("Authorization", "Bearer "+slr.client.GetSid())
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		buf := new(bytes.Buffer)
+		buf.ReadFrom(resp.Body)
+		return nil, fmt.Errorf("ERROR: statuscode: %d, body: %s", resp.StatusCode, buf.String())
+	}
```

Comment:
How about adding a retry mechanism when the code is 5xx?
https://github.com/avast/retry-go